GVFS/GVFS.Common/Maintenance/PackfileMaintenanceStep.cs

```diff
@@ -0,0 +1,122 @@
+
+        // public only for unit tests
+        public void CountPackFiles(out int count, out long size)
+        {
+            count = 0;
+            size = 0;
+
+            foreach (DirectoryItemInfo info in this.Context.FileSystem.ItemsInDirectory(this.Context.Enlistment.GitPackRoot))
+            {
+                if (string.Equals(Path.GetExtension(info.Name), ".pack", StringComparison.OrdinalIgnoreCase))
+                {
+                    count++;
+                    size += info.Length;
+                }
+            }
+        }
+
+        // public only for unit tests
+        public IEnumerable<string> CleanStaleIdxFiles()
+        {
+            List<DirectoryItemInfo> packDirContents = this.Context
+                                                          .FileSystem
+                                                          .ItemsInDirectory(this.Context.Enlistment.GitPackRoot)
+                                                          .ToList();
+
+            foreach (DirectoryItemInfo info in packDirContents)
+            {
+                if (string.Equals(Path.GetExtension(info.Name), ".idx", StringComparison.OrdinalIgnoreCase))
+                {
+                    string pairedPack = info.FullName.Substring(0, info.FullName.Length - "idx".Length) + "pack";
+
+                    if (!this.Context.FileSystem.FileExists(pairedPack))
+                    {
+                        if (this.Context.FileSystem.TryDeleteFile(info.FullName))
+                        {
+                            yield return info.Name;
+                        }
+                    }
+                }
+            }
+        }
+
+        protected override void PerformMaintenance()
+        {
+            using (ITracer activity = this.Context.Tracer.StartActivity(this.Area, EventLevel.Informational, Keywords.Telemetry, metadata: null))
+            {
+                // forceRun is only currently true for functional tests
+                if (!this.forceRun)
+                {
+                    if (!this.EnoughTimeBetweenRuns())
+                    {
+                        activity.RelatedWarning($"Skipping {nameof(LooseObjectsStep)} due to not enough time between runs");
+                        return;
+                    }
+
+                    IEnumerable<int> processIds = this.RunningGitProcessIds();
+                    if (processIds.Any())
+                    {
+                        activity.RelatedWarning($"Skipping {nameof(LooseObjectsStep)} due to git pids {string.Join(",", processIds)}", Keywords.Telemetry);
+                        return;
+                    }
+                }
+
+                this.CountPackFiles(out int beforeCount, out long beforeSize);
+                GitProcess.Result result = this.RunGitCommand((process) => process.MultiPackIndexExpire(this.Context.Enlistment.GitObjectsRoot));
+                List<string> staleIdxFiles = this.CleanStaleIdxFiles().ToList();

```

Comment:
From the explaination on the header it seems like expire should have removed the idx's but may not have since they could have a handle open. Is that correct.  A line here explaining would be great.